pkix/csr.go

```diff
@@ -52,14 +52,16 @@ var (
 )
 
 // ParseAndValidateIPs parses a comma-delimited list of IP addresses into an array of IP addresses
-func ParseAndValidateIPs(ipList string) (res []net.IP) {
+func ParseAndValidateIPs(ipList string) (res []net.IP, err error) {
 	ips := strings.Split(ipList, ",")
 	for _, ip := range ips {
-		parsedIP := net.ParseIP(ip)
-		if parsedIP == nil {
-			return nil
+		if len(ip) > 0 {
+			parsedIP := net.ParseIP(ip)
+			if parsedIP == nil {
+				return nil, errors.New(ip)
```

Comment:
nit: you could just return `errors.New(fmt.Sprintf("invalid IP address %", ip))` here instead of having one piece of the error message crafted here and another in the caller.